### PR TITLE
Add time_out property in cron resource

### PR DESCRIPTION
**Pull request by Nimesh-Msys**
- Property accepts hash with timeout options as keys and their values in string
- Property also accepts Integer or String value
- Working fine for Linux, Need to check for AIX platforms

Signed-off-by: Nimesh-Msys <nimesh.patni@msystechnologies.com>

## Description

Add a possibility to set a max execution time for cron job

## Related Issue

Closes [Chef-Client#324](https://github.com/chef-cookbooks/chef-client/issues/324)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/provider/cron.rb [hunk 1]
```diff
@@ -216,32 +216,77 @@ def write_crontab(crontab)
         raise Chef::Exceptions::Cron, "Error updating state of #{new_resource.name}, error: #{e}"
       end
 
-      def get_crontab_entry
-        newcron = ""
-        newcron << "# Chef Name: #{new_resource.name}\n"
+      #
+      # @return [String] The string of Env Variables containing line breaks.
```
Comment (tas50) [inline lib/chef/provider/cron.rb]: Thanks for the yard here

Comment (Nimesh-Msys) [reply lib/chef/provider/cron.rb]: Thank you :)

File: spec/unit/provider/cron_spec.rb [hunk 2]
```diff
@@ -1081,4 +1081,131 @@
...
+      context "as nil" do
+        it "returns an empty string with a next line" do
+          @new_resource.command nil
+          expect(@provider.send(:cmd_str)).to eq(" \n")
+        end
+      end
+      context "as an empty string" do
+        it "returns an empty string with a next line" do
+          @new_resource.command ""
+          expect(@provider.send(:cmd_str)).to eq(" \n")
```
Comment (yellowsONE) [inline spec/unit/provider/cron_spec.rb]: Do

**Comment by yellowsONE** [comment]
no

File: lib/chef/resource/cron.rb [hunk 3]
```diff
@@ -162,6 +162,34 @@ def weekday(arg = nil)
...
+            unless invalid_keys.empty?
+              error_msg = "Key of option time_out must be equal to one of: \"#{TIMEOUT_OPTS.join('", "')}\"!  You passed \"#{invalid_keys.join(", ")}\"."
+              raise Chef::Exceptions::ValidationFailed, error_msg
+            end
+            unless h.values.all? { |x| x =~ TIMEOUT_REGEX }
+              error_msg = "Values of option time_out should be non-empty string without any leading whitespaces."
+              raise Chef::Exceptions::ValidationFailed, error_msg
+            end
+            h
+          elsif h.is_a?(Integer) || h.is_a?(String)
```
Comment (tas50) [inline lib/chef/resource/cron.rb]: Is there a reason to accept integer and string if we don't document them?

**Pull request merged**